openapi.yaml

```diff
@@ -0,0 +1,191 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Group'
+  /person/{id}:
+    get:
+      summary: requesting a person
+      operationId: getPerson
+      parameters:
+        - $ref: '#/components/parameters/id'
+      responses:
+        200:
+          description: returns a person
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Person'
+    put:
+      summary: update a person
+      operationId: updatePerson
+      requestBody:
+        $ref: '#/components/requestBodies/person'
+      parameters:
+        - $ref: '#/components/parameters/id'
+      responses:
+        200:
+          description: successfully updated person
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Person'
+  /person:
+    post:
+      summary: creating a person
+      operationId: createPerson
+      requestBody:
+        $ref: '#/components/requestBodies/person'
+      responses:
+        201:
+          description: successfully created person
+          content:
+            application/json:
+              schema:
+                allOf:
+                  - $ref: '#/components/schemas/Person'
+                  - type: object
+                    properties:
+                      id:
+                        $ref: '#/components/schemas/uuid'
+components:
+  requestBodies:
+    group:
+      required: true
+      description: name of a group
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              name:
+                type: string
+    person:
+      required: true
+      description: a person
+      content:
+        application/json:
+          schema:
+            type: object
+            properties:
+              surname:
+                type: string
+              name:
+                type: string
+              major:
+                $ref: '#/components/schemas/Major'
+              gender:
+                $ref: '#/components/schemas/Gender'
+              age:
+                minimum: 0
+                type: integer
+  parameters:
+    id:
+      in: path
+      name: id
+      required: true
+      schema:
+        $ref: '#/components/schemas/uuid'
+      description: the unique identifier for the resource
+  schemas:
+    uuid:
+      format: "$[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}^"
+      type: string
+      description: a unique identifier, in UUID format
+      example: 00000000-0000-0000-0000-000000000000
+    Person:
+      properties:
+        id:
+          $ref: '#/components/schemas/uuid'
+        groupId:
+          $ref: '#/components/schemas/uuid'
+        surname:
+          type: string
+        name:
+          type: string
+        major:
+          $ref: '#/components/schemas/Major'
```

Comment:
```suggestion
          $ref: '#/components/schemas/major'
```

Reply:
The openAPI generator for the golang server doesn't like it when the schema names start with a lower-case letter.